File: insertion sort list.cpp
class Solution {
public:
    ListNode* insertionSortList(ListNode* head) {
    int counter=0;
    ListNode *temp=head; 
    int x;
    while (temp->next!=NULL)
    {
    counter++;
    temp=temp->next;
    }
    for(int i=1;i<=counter;i++)
    {
        temp=head;
        while (temp->next!=NULL)
        {
           if(temp->val > temp->next->val)
           {
            x=temp->val;
            temp->val=temp->next->val;
            temp->next->val=x;
           }
           temp=temp->next;
        }
    }
    return head;
    }
};

File: pre-camp/week/leetcode/insertion-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        temp = head
        while temp.next:
            count+=1
            temp = temp.next
        for i in range(count):
            temp = head 
            while temp.next:
                if temp.val > temp.next.val:
                    x= temp.val
                    temp.val = temp.next.val
                    temp.next.val = x
                temp= temp.next
        return head

